CHANGELOG.md

```diff
@@ -17,12 +17,15 @@ All notable changes to this project will be documented in this file.
 - [#1112](https://github.com/pints-team/pints/pull/1112) A new `NoUTurnMCMC` sampler (NUTS) was added, along with a `DualAveragingAdaption` class to adaptively tune related Hamiltonian Monte Carlo methods.
 - [#1025](https://github.com/pints-team/pints/pull/1025) Added a stochastic logistic growth problem for use with ABC.
 ### Changed
+- [#1250](https://github.com/pints-team/pints/pull/1250) The returned values from `SingleChainMCMC.tell()` and `MultiChainMCMC.tell()` have been extended from current position `x` to `x, fx, accepted`, where `fx` is the current log likelihood and `accepted` is a bool indicating whether tell performed an acceptance step in this call.
```

Comment:
Still thinking be best to return a dict for `fx` and `accepted`... I don't know what else we'd ever want to return in the tell, but since we *encourage* others to use the ask-tell methods, we probably don't want to make changes to the API. I thought _maybe_ having it as a dict would minimise the chance of changing the API even if we want to add something more to be returned in `tell`...

Because I'm imaging I'd be doing
```
x, fx, accepted = mcmc.tell()
```
and if we add another one output to it, then the code above breaks (not enough value to unpack - as I need `x, fx, accepted, _` to unpack). But if we return a dictionary, then we'd ask others to do something like
```
next = mcmc.tell()
x = next['x']
fx = next['fx']
accepted = next['accepted']
```
and even if we add another key (I don't know what, but e.g.) `fx2` to it, the above code would still work.

Not sure, what do you think?

Reply:
I really don't want to return dicts from tell.
It would mean lots of unnecessary dict creation and dict lookups, in a method we're expecting to call 10000s of times.

And the fact that `x, fx, accepted = ...` would break is a feature. It means we have a rigid interface that people need to stick to, so that everyone knows what to expected. Otherwise we need to make the whole "contract" for using ask-tell a much softer thing that the user needs to check instead of having the programming language check it.

To me those downsides are much bigger than the possible benefit of having some future extension of `tell()` being slightly easier.

Reply:
And we've found that `x`, `fx`, and `accepted` are used by the controller and in some meta-methods, so there's a clear case for them being there.
And we've found that they exist and make sense for each method so far, so a good case for having them being solid variables rather than things that might exist in a dict